Samples/ShareTarget/cs/ShareTarget.xaml.cs

```diff
@@ -286,6 +286,25 @@ public ShareTarget()
                         {
                             AddContentValue("HTML: ");
                             ShareWebView.Visibility = Visibility.Visible;
+                            // Check if there are any local images in the resource map.
+                            if (this.sharedResourceMap.Count > 0)
+                            {
+                                foreach (KeyValuePair<string, RandomAccessStreamReference> item in this.sharedResourceMap)
+                                {
+                                    var stream = await item.Value.OpenReadAsync();
+                                    if (!stream.ContentType.Contains("image/"))
+                                        continue;
+                                    var reader = new DataReader(stream.GetInputStreamAt(0));
+                                    await reader.LoadAsync((uint)stream.Size);
+                                    byte[] byteArray = new byte[stream.Size];
+                                    reader.ReadBytes(byteArray);
+                                    String base64String = "<img src ='data:image/gif;base64," + Convert.ToBase64String(byteArray) + "'";
+                                    String replaceTarget = "<img src=" + item.Key;
+
+                                    htmlFragment = htmlFragment.Replace(replaceTarget, base64String);
```

Comment:
Not entirely sure that was what you want.

I also have an alternate solution to this if this wasn't what you wanted.


                        if (!String.IsNullOrEmpty(htmlFragment))
                        {
                            if (this.sharedResourceMap.Count > 0)
                            {
                                foreach (KeyValuePair<string, RandomAccessStreamReference> item in this.sharedResourceMap)
                                {
                                    try
                                    {
                                        using (var resourceStream = await item.Value.OpenReadAsync())
                                        {
                                            var tempFile = await folder.CreateFileAsync(Path.GetFileName(item.Key), CreationCollisionOption.OpenIfExists);
                                            using (var stream = await tempFile.OpenStreamForWriteAsync())
                                            {
                                                await resourceStream.AsStreamForRead().CopyToAsync(stream);
                                            }
                                            htmlFragment = htmlFragment.Replace(item.Key, tempFile.Name);
                                        }
                                    }
                                    catch (Exception ex)
                                    {
                                        htmlFragment = htmlFragment.Replace(item.Key, @"ms-appx-web:///Assets/error.png");
                                    }
                                }
                            }

                            AddContentValue("HTML: ");
                            ShareWebView.Visibility = Visibility.Visible;

                            var html = "<html><body>" + htmlFragment + "</body></html>";

                            var file = await folder.CreateFileAsync("html.html", CreationCollisionOption.OpenIfExists);
                            await FileIO.WriteTextAsync(file, html);

                            ShareWebView.Navigate(new Uri("ms-appdata:///temp/folder/html.html"));
                        }